bemuse/src/game/loaders/samples-loader.js

```diff
@@ -20,40 +21,37 @@ export class SamplesLoader {
         })
       })
     if (decodeProgress) decodeProgress.formatter = EXTRA_FORMATTER
-    return Promise.map(files, load, { concurrency: 64 }).then((arr) =>
-      _(arr).filter().fromPairs().value()
-    )
+    return PromisePool.withConcurrency(64)
+      .for(files)
+      .process(load)
+      .then(({ results: arr }) => _(arr).filter().fromPairs().value())
```

Comment:
`PromisePool`’s behavior is different from `Promise.map` that when it encounters an error, the promise does not get rejected. It is quite worrying because errors that happened inside the pool will now be silently discarded.

A closer alternative to `Promise.map` would be the [`p-map` npm package](https://www.npmjs.com/package/p-map) which supports the same `concurrency` option.

Reply:
Fixed on e69c841c. Thanks for information.

Reply:
@MikuroXina You can downgrade `p-map` to v4.0.0, that may keep the scope of this PR small.

Reply:
I tried that but it seems that I mistook to downgrade it. I forgot that `bemuse-indexer` also depends it. Fixed on 7302ac95.

Reply:
Hmm, it fails to build because `aggregate-error` dependency from `p-map` is using private class field feature. I will try to fix the config.